homeassistant/components/reolink/switch.py

```diff
@@ -0,0 +1,106 @@
+"""This component provides support for Reolink switch entities."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+from reolink_aio.api import Host
+
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ReolinkData
+from .const import DOMAIN
+from .entity import ReolinkCoordinatorEntity
+
+
+@dataclass
+class ReolinkSwitchEntityDescriptionMixin:
+    """Mixin values for Reolink switch entities."""
+
+    value: Callable[[Host, int | None], bool]
+    method: Callable[[Host, int | None, bool], Any]
+
+
+@dataclass
+class ReolinkSwitchEntityDescription(
+    SwitchEntityDescription, ReolinkSwitchEntityDescriptionMixin
+):
+    """A class that describes switch entities."""
+
+    supported: Callable[[Host, int | None], bool] = lambda api, ch: True
+
+
+SWITCH_ENTITIES = (
+    ReolinkSwitchEntityDescription(
+        key="floodlight",
+        name="Floodlight",
+        icon="mdi:spotlight-beam",
+        supported=lambda api, ch: api.supported(ch, "floodLight"),
+        value=lambda api, ch: api.whiteled_state(ch),
+        method=lambda api, ch, value: api.set_whiteled(ch, state=value),
+    ),
+    ReolinkSwitchEntityDescription(
+        key="ir_lights",
+        name="Infra red lights",
+        icon="mdi:led-off",
+        supported=lambda api, ch: api.supported(ch, "ir_lights"),
+        value=lambda api, ch: api.ir_enabled(ch),
+        method=lambda api, ch, value: api.set_ir_lights(ch, value),
+    ),
```

Comment:
Same as above?

Reply:
How is this complicated for a light and not a switch? Both are toggle entities...

Reply:
I have removed the IR LEDs switch in favor of a light entity in this PR: https://github.com/home-assistant/core/pull/88619
I have also renamed the entity to "Infra red lights in night mode"